app/forms/search_developer_form.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class SearchDeveloperForm
+  include Virtus.model
+  include ActiveModel::Model
+
+  attribute :programming_language, ProgrammingLanguage
+
+  def search
+    query = Developer.all
+    return query if programming_language.empty?
+
+    query = query.includes(:developer_programming_languages)
+                 .joins(:developer_programming_languages)
+                 .where(
+                   'developer_programming_languages.programming_language_id=?',
+                   programming_language
```

Comment:
- `where(developer_programming_languages: { programming_language_id: programming_language })` is slightly better.
- I expect `programming_language` is a string (e.g. `Ruby`). But it's actually `id`, right? Then it should be  `programming_language_id`.